src/pages/PremiumVideoCall/components/PreJoinScreens/DeviceSelectionScreen/DeviceSelectionScreen.tsx

```diff
@@ -66,20 +68,32 @@ interface DeviceSelectionScreenProps {
     name: string
     roomName: string
     setStep: (step: Steps) => void
+    token: string
 }
 
-export default function DeviceSelectionScreen({
+function DeviceSelectionScreen({
     name,
     roomName,
     setStep,
+    token,
 }: DeviceSelectionScreenProps) {
     const classes = useStyles()
-    const { getToken, isFetching } = useAppState()
+    const { isFetching } = useAppState()
     const { connect, isAcquiringLocalTracks, isConnecting } = useVideoContext()
     const disableButtons = isFetching || isAcquiringLocalTracks || isConnecting
 
     const handleJoin = () => {
-        getToken(name, roomName).then((token) => connect(token))
+        const api = new Api(token)
+        api.createRoom()
+            .then((response) => response.roomId)
+            .then((roomId) => {
+                return api.joinRoom(roomId)
+            })
+            .then((response) => {
+                const { jwtAccessToken } = response
+                return jwtAccessToken
+            })
+            .then((jwtAccessToken) => connect(jwtAccessToken))
```

Comment:
This code will change in order to only use `joinRoom`, once we merge @HugoGresse PR. I guess I can replace with `async/await` at that moment